base/strings/basic.jl

```diff
@@ -135,17 +135,44 @@ end
 
 ## Generic indexing functions ##
 
-prevind(s::DirectIndexString, i::Integer) = i-1
-prevind(s::AbstractArray   , i::Integer) = i-1
-nextind(s::DirectIndexString, i::Integer) = i+1
-nextind(s::AbstractArray   , i::Integer) = i+1
+prevind(s::DirectIndexString, i::Integer) = convert(Int, i)-1
+prevind(s::AbstractArray   , i::Integer) = convert(Int, i)-1
+nextind(s::DirectIndexString, i::Integer) = convert(Int, i)+1
+nextind(s::AbstractArray   , i::Integer) = convert(Int, i)+1
+
+prevind(s::String, i::Integer) = prevind(s, convert(Int, i))
+nextind(s::String, i::Integer) = nextind(s, convert(Int, i))
+
+function prevind(s::String, i::Int)
```

Comment:
It's not ideal that the generic method below accepts any `Integer`, but this one only accepts `Int`. This means the generic version will be used for non-`Int`, for no good reason. Could you loosen the signature, and use `j = Int(i)` as in the other one?